internal/controllers/topology/cluster/cluster_controller.go

```diff
@@ -349,3 +369,22 @@ func dryRunPatchHelperFactory(c client.Client) structuredmerge.PatchHelperFactor
 		return structuredmerge.NewTwoWaysPatchHelper(original, modified, c, opts...)
 	}
 }
+
+func (r *Reconciler) requestIsRateLimited(ctx context.Context, req reconcile.Request) (bool, error) {
+	log := ctrl.LoggerFrom(ctx)
+
+	// If no rateLimitingCache is defined return false.
+	if r.rateLimitingCache == nil {
+		return false, nil
+	}
+	if _, found, err := r.rateLimitingCache.Get(req.String()); found && err == nil {
+		log.Info(fmt.Sprintf("Requeuing %s due to rate limit", req.String()))
+		return true, nil
+	} else if err != nil {
+		return false, errors.Wrap(err, fmt.Sprintf("failed to read item %s from rate limiting cache", req.String()))
+	}
+	if err := r.rateLimitingCache.Add(req.String()); err != nil {
```

Comment:
```suggestion
   // return false but start tracking the object has been processed to avoid further reconcile until TTL expires.
	if err := r.rateLimitingCache.Add(req.String()); err != nil {
```